handlers.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+func HandlePokemonById(rw http.ResponseWriter, r *http.Request, ps []Pokemon) {
+
+	vars := mux.Vars(r)
+	pokemonId, err := strconv.Atoi(vars["id"])
+
+	if err != nil {
+		rw.WriteHeader(http.StatusBadRequest)
+		rw.Write([]byte("Id passed is not a number"))
+		return
+	}
+
+	for _, p := range ps {
+		if p.Id == pokemonId {
+
+			jsonResponse, err := json.Marshal(p)
+
+			if err != nil {
+				return
+			}
+
+			rw.Write(jsonResponse)
+		}
+	}
+
```

Comment:
If you weren't able to get a pokemon, then you would like to let the user know that, it would be appropriate to return a NotFound status code with a message that could probably say `Pokemon not found` or `Pokemon doesn't exist`